pkg/segment/writer/packer.go

```diff
@@ -672,6 +680,38 @@ func encJsonNumber(key string, numType SS_IntUintFloatTypes, intVal int64, uintV
 	return valSize
 }
 
+func (ss *SegStore) updateColValueSizeInAllSeenColumns(colName string, size uint32) {
+	currentSize, ok := ss.AllSeenColumns[colName]
+	if !ok {
+		if ss.RecordCount > 0 {
+			// Not the first record in the segment, so the col value size is not consistent.
+			ss.AllSeenColumns[colName] = -1
+			currentSize = -1
+		} else {
+			ss.AllSeenColumns[colName] = 0
+			currentSize = 0
+		}
+	}
+
+	if currentSize == -1 {
+		return
+	}
+
+	if size > math.MaxInt8 {
+		ss.AllSeenColumns[colName] = -1
+		return
+	}
+
+	if currentSize == 0 {
+		ss.AllSeenColumns[colName] = int8(size)
+		return
+	}
+
+	if currentSize != int8(size) {
```

Comment:
would be great if we can avoid type conversions, since this is the fast path code, understand the need of -1, but we can pick some other value for noting down inconsistency

Reply:
Then I think I can make the type as `uint32` and for inconsistency the value will be `0`.  As the size by default comes in `uint32`.